app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,24 @@
+class FavoritesController < ApplicationController
+  before_action :logged_in_user
+
+  def create
+    @micropost = Micropost.find(params[:micropost_id])
+
+    current_user.favor(@micropost)
+    respond_to do |format|
+      format.html { redirect_back_or @micropost.user }
+      format.js
+    end
+  end
+
+  def destroy
+    fav = Favorite.find_by_id(params[:id])
```

Comment:
`find(params[:id])`だと、該当の`favorite`が無いときに例外出ちゃうので。
[9dc4950](https://github.com/Joe-noh/sample_app/commit/9dc4950e3fcfd43a1fec680860883c4e3a3d1b16)に書いた黙殺をやるために、見つからなかったら`nil`を返す`find_by_id`使いました。

今調べたら`find_by(:id, ...)`も`nil`返してくれるようなので、こっちでもいいと思います。
でもそうなると、`find_by_id`は何者なのかという疑問が出てきたので調べたら
mildly deprecatedとか書いてありました。

>  # == Dynamic attribute-based finders
>  #
>  # Dynamic attribute-based finders are a mildly deprecated way of getting (and/or creating) objects
>  # by simple queries without turning to SQL. They work by appending the name of an attribute
>  # to <tt>find_by_</tt> like <tt>Person.find_by_user_name</tt>.
>  # Instead of writing <tt>Person.find_by(user_name: user_name)</tt>, you can use
>  # <tt>Person.find_by_user_name(user_name)</tt>.

`find_by(id: ...)`が最適だと思います。